### ISSUE TITLE
Proposed Rule: No arrow function in conditional block.

### ISSUE THREAD

**Issue by leebyron** [eslint/eslint#3280]
http://eslint.org/docs/rules/arrow-parens highlights the potential typo of writing `=>` when you meant `>=` in conditional bodies.

I would prefer a rule which explicitly tackles this particular issue regardless of if the arrow function parameters are wrapped in parens.

Examples:

```js
if (a => a) { // fail: arrow function in if statement.
} else if ((a) => a) { // fail: arrow function in if statement.
}

while(a => a) { // fail
}

for (var i = 0; i => 10; i++) { // fail
}
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by michaelficarra** [comment]
:+1: This could catch some nasty typos.

**Comment by mysticatea** [comment]
FYI: There is [a plugin](https://github.com/Jxck/eslint-plugin-arrow-function). Jxck is author of `arrow-parens`/`arrow-spacing`.

This rule looks good to me.

**Comment by nzakas** [comment]
Makes sense. Maybe `no-arrow-condition`?

**Comment by Jxck** [comment]
welcome to PR my plugin code for this rule :)

**Comment by lukekarrys** [comment]
I'm working on this.

**Comment by lukekarrys** [comment]
I created #4187 for the addition of this rule, which includes an option, if anyone wants to discuss over there.
